### PR TITLE
feat(web): Remove Sign In link and set up admin login

### PR DESCRIPTION
**Pull request by ERosendo**
This PR removes the Sign In link on the header, adds django-ratelimit as a dependency, and redirects the admin login to a custom login page that uses a ratelimiting decorator. 

Here's a screenshot of the login page:

image

### CODE REVIEW

File: bc/core/utils/network.py [hunk 1]
```diff
@@ -18,6 +20,13 @@ def strip_port_to_make_ip_key(ip_str: str | None) -> str | None:
     :param ip_str: the IP address of the viewer and the source port of the request
```
Comment (mlissner) [inline bc/core/utils/network.py:20]: The docstring needs a few tweaks here.

File: bc/users/templates/register/login.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <h1 class="font-display font-light tracking-wide">Sign In</h2>
+    {% if form.errors %}
+        {%  for error in form.non_field_errors %}
+        <p class="alert alert-danger">{{ error|safe }}</p>
+        {% endfor %}
+    {% endif %}
+
+    <form action="{% url "sign-in" %}" method="post">
+        {% csrf_token %}
+    <div class="grid grid-cols-1 gap-4">
```
Comment (mlissner) [inline bc/users/templates/register/login.html:17]: Indentation is a bit weird on this div. Also, we should be doing two spaces, not four in HTML, according to our editorconfig.

File: bc/users/urls.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+urlpatterns = [
+    # Sign in page
+    path(
+        "sign-in/",
+        ratelimiter_unsafe_10_per_m(
+            auth_views.LoginView.as_view(
+                **{
+                    "template_name": "register/login.html",
+                    "extra_context": {"private": False},
```
Comment (mlissner) [inline bc/users/urls.py:14]: We're not using this yet in our base template, are we? If not, we should either remove it from here or add it to the base template. I think we can probably just remove it here and we won't have private pages in this project?

**Review by mlissner — CHANGES_REQUESTED** [review]
Good stuff. Looks good too. A few comments, thank you!

File: bc/users/urls.py (hunk 3 above)
Comment (ERosendo) [reply bc/users/urls.py]: No, We're not using the "private" attribute in the base.html. I'll remove it.

File: bc/core/utils/network.py (hunk 1 above)
Comment (ERosendo) [reply bc/core/utils/network.py]: Thanks!. I'm going to update the docstring.

**Review by mlissner — APPROVED** [review]

**Pull request merged**